widget/embedded/src/pages/SelectSwapItemsPage.tsx

```diff
@@ -47,8 +48,12 @@ export function SelectSwapItemsPage(props: PropTypes) {
     blockchain: selectedBlockchainName,
     searchFor: searchedFor,
   });
-  const tokensList = sortTokensByBalance(tokens, getBalanceFor);
+  const pinnedTokens = tokens.filter((token) => isTokenPinned(token, type));
+  const otherTokens = tokens.filter((token) => !isTokenPinned(token, type));
 
+  const sortedOtherTokens = sortTokensByBalance(otherTokens, getBalanceFor);
+  const sortedPinnedTokens = sortTokensByBalance(pinnedTokens, getBalanceFor);
+  const tokensList = [...sortedPinnedTokens, ...sortedOtherTokens];
```

Comment:
please move all sorting logic to sortTokensByBalance function (Change its name to something more meaningful) and avoid unnecessary iterations.

Reply:
fixed